### PR TITLE
[Refactor]All agent tasks use new thread pool

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/agent/agent_server.cpp [hunk 1]
```diff
@@ -344,42 +367,51 @@ void AgentServer::Impl::submit_tasks(TAgentResult& agent_result, const std::vect
             break;
         }
         case TTaskType::CLEAR_TRANSACTION_TASK:
-            _clear_transaction_task_workers->submit_tasks(all_tasks);
+            HANDLE_TASK(TTaskType::CLEAR_TRANSACTION_TASK, all_tasks, run_clear_transaction_task,
+                        ClearTransactionAgentTaskRequest, clear_transaction_task_req, nullptr);
             break;
         case TTaskType::CLONE:
-            _clone_workers->submit_tasks(all_tasks);
+            HANDLE_TASK(TTaskType::CLONE, all_tasks, run_clone_task, CloneAgentTaskRequest, clone_req, nullptr);
```
Comment (trueeyu) [inline be/src/agent/agent_server.cpp:374]: May be crash if `env=nullptr`?

File: be/src/agent/agent_server.cpp [hunk 2]
```diff
@@ -452,25 +484,37 @@ ThreadPool* AgentServer::Impl::get_thread_pool(int type) const {
...
+        return _thread_pool_download.get();
     case TTaskType::MOVE:
+        return _thread_pool_move_dir.get();
+    case TTaskType::UPDATE_TABLET_META_INFO:
+        return _thread_pool_update_tablet_meta_info.get();
+    case TTaskType::ALTER:
+        return _thread_pool_alter_tablet.get();
+    case TTaskType::CLEAR_TRANSACTION_TASK:
+        return _thread_pool_clear_transaction.get();
+    case TTaskType::PUSH:
```
Comment (trueeyu) [inline be/src/agent/agent_server.cpp:508]: Will `PUSH` use the `delete_pool`?

File: be/src/service/service_be/lake_service.cpp [hunk 3]
```diff
@@ -317,33 +336,47 @@ void LakeServiceImpl::get_tablet_stats(::google::protobuf::RpcController* contro
...
-            LOG(WARNING) << "Fail to get tablet " << tablet_id << ": " << tablet.status();
-            continue;
-        }
+        auto task = [&, tablet_info]() {
+            int64_t tablet_id = tablet_info.tablet_id();
+            auto tablet = _env->lake_tablet_manager()->get_tablet(tablet_id);
+            if (!tablet.ok()) {
+                LOG(WARNING) << "Fail to get tablet " << tablet_id << ": " << tablet.status();
+                latch.count_down();
+                return;
```
Comment (trueeyu) [inline be/src/service/service_be/lake_service.cpp:348]: continue?

File: be/src/service/service_be/lake_service.cpp [hunk 4]
```diff
@@ -288,21 +289,39 @@ void LakeServiceImpl::delete_data(::google::protobuf::RpcController* controller,
...
-            continue;
-        }
+        auto task = [&, tablet_id]() {
+            auto tablet = _env->lake_tablet_manager()->get_tablet(tablet_id);
+            if (!tablet.ok()) {
+                LOG(WARNING) << "Fail to get tablet " << tablet_id << ": " << tablet.status();
+                std::lock_guard l(response_mtx);
+                response->add_failed_tablets(tablet_id);
+                latch.count_down();
+                return;
```
Comment (trueeyu) [inline be/src/service/service_be/lake_service.cpp:303]: continue?

File: be/src/agent/agent_task.cpp [hunk 5]
```diff
@@ -5,24 +5,158 @@
...
+    }
+    return status;
+}
+
+static void alter_tablet(const TAlterTabletReqV2& agent_task_req, int64_t signature, TTaskType::type task_type,
+                         TFinishTaskRequest* finish_task_request) {
+    AgentStatus status = STARROCKS_SUCCESS;
+    TStatus task_status;
+    std::vector<std::string> error_msgs;
+
```
Comment (trueeyu) [inline be/src/agent/agent_task.cpp:46]: How to process the timeout task?

File: be/src/service/service_be/lake_service.cpp [hunk 6]
```diff
@@ -288,21 +289,39 @@ void LakeServiceImpl::delete_data(::google::protobuf::RpcController* controller,
         return;
     }
 
+    auto thread_pool = _env->agent_server()->get_thread_pool(TTaskType::DROP);
+    auto latch = BThreadCountDownLatch(request->tablet_ids_size());
+    bthread::Mutex response_mtx;
```
Comment (trueeyu) [inline be/src/service/service_be/lake_service.cpp:294]: Why add the lock?

File: be/src/service/service_be/lake_service.cpp [hunk 7]
```diff
@@ -317,33 +336,47 @@ void LakeServiceImpl::get_tablet_stats(::google::protobuf::RpcController* contro
...
-        for (const auto& rowset : (*tablet_metadata)->rowsets()) {
-            num_rows += rowset.num_rows();
-            data_size += rowset.data_size();
+            int64_t num_rows = 0;
+            int64_t data_size = 0;
+            for (const auto& rowset : (*tablet_metadata)->rowsets()) {
+                num_rows += rowset.num_rows();
+                data_size += rowset.data_size();
+            }
+            auto tablet_stat = response->add_tablet_stats();
```
Comment (trueeyu) [inline be/src/service/service_be/lake_service.cpp:366]: Why not add lock?

**Comment by wangruin** [comment]
run starrocks_be_unittest

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/service/service_be/lake_service.cpp (hunk 7 above)
Comment (trueeyu) [reply be/src/service/service_be/lake_service.cpp:386]: If there is no lock here, is it thread safe?

**Review by trueeyu — DISMISSED** [review]

File: be/src/service/service_be/lake_service.cpp [hunk 8]
```diff
@@ -434,11 +512,22 @@ void LakeServiceImpl::upload_snapshots(::google::protobuf::RpcController* contro
...
+        auto st = loader->upload(request);
+        if (!st.ok()) {
+            cntl->SetFailed(st.to_string());
+        }
+        latch.count_down();
+    };
+    auto st = thread_pool->submit_func(task);
     if (!st.ok()) {
-        cntl->SetFailed(st.to_string());
+        LOG(WARNING) << "Fail to submit upload snapshots task: " << st;
```
Comment (sduzh) [inline be/src/service/service_be/lake_service.cpp:527]: `cntl->SetFailed(st.to_string())`

File: be/src/service/service_be/lake_service.cpp [hunk 9]
```diff
@@ -408,18 +474,30 @@ void LakeServiceImpl::unlock_tablet_metadata(::google::protobuf::RpcController*
...
+            cntl->SetFailed("Fail to unlock tablet metadata");
+        }
+        latch.count_down();
+    };
+    auto st = thread_pool->submit_func(task);
     if (!st.ok()) {
-        LOG(ERROR) << "Fail to unlock tablet metadata, tablet id: " << request->tablet_id()
-                   << ", version: " << request->version();
-        cntl->SetFailed("Fail to unlock tablet metadata");
+        LOG(WARNING) << "Fail to submit unlock tablet metadata task: " << st;
```
Comment (sduzh) [inline be/src/service/service_be/lake_service.cpp:497]: cntl->SetFailed()

File: be/src/service/service_be/lake_service.cpp [hunk 10]
```diff
@@ -366,27 +419,40 @@ void LakeServiceImpl::lock_tablet_metadata(::google::protobuf::RpcController* co
...
+        }
+        latch.count_down();
+    };
+    auto st = thread_pool->submit_func(task);
     if (!st.ok()) {
-        LOG(ERROR) << "Fail to lock tablet metadata, tablet id: " << request->tablet_id()
-                   << ", version: " << request->version();
-        cntl->SetFailed("Fail to lock tablet metadata");
-        return;
+        LOG(WARNING) << "Fail to submit lock tablet metadata task: " << st;
```
Comment (sduzh) [inline be/src/service/service_be/lake_service.cpp:451]: cntl->SetFailed

**Comment by abc982627271** [comment]
run starrocks_be_unittest

**Comment by abc982627271** [comment]
run all

**Review by sduzh — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**